src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -49,22 +48,23 @@ class TezosPlatformDiscoveryOperations(
   networkName: String = "notUsed")
   (implicit executionContext: ExecutionContext) {
 
+  import MetadataCaching._
   import cats.effect._
   import cats.implicits._
 
 
   /** Method for initializing values of the cache */
   def init(): Future[Unit] = {
-    val result = for {
-      ent <- IO.fromFuture(IO(metadataOperations.runQuery(preCacheEntities)))
-      _ <- caching.putAllEntitiesIntoEmptyCache(ent)
-      attr <- IO.fromFuture(IO(metadataOperations.runQuery(preCacheAttributes)))
-      _ <- caching.putAllAttributesIntoEmptyCache(attr)
-      attrValues <- IO.fromFuture(IO(metadataOperations.runQuery(preCacheAttributeValues)))
-      _ <- caching.putAllAttributeValuesIntoEmptyCache(attrValues)
-    } yield ()
-
-    result.unsafeToFuture()
+    val entities = IO.fromFuture(IO(metadataOperations.runQuery(preCacheEntities)))
+    val attributes = IO.fromFuture(IO(metadataOperations.runQuery(preCacheAttributes)))
+    val attributeValues = IO.fromFuture(IO(metadataOperations.runQuery(preCacheAttributeValues)))
+
+    (entities, attributes, attributeValues).mapN { case (ent, attr, attrValues) =>
+      (caching.fillEntitiesCache(ent), caching.fillAttributesCache(attr), caching.fillAttributeValuesCache(attrValues))
+        .mapN {
```

Comment:
I didn't realize at first that the operations were bound in paired locksteps.
Maybe it would read better if the dependent operations were actually flatMapped as needed
i.e.
```scala
(entities flatMap caching.fillEntitiesCache, attributes flatMap caching.fillAttributesCache, attributeValues flatMap caching.fillAttributeValuesCache)
  .tuple3 //same as mapN (possibly there's even a more generic .tuple) but with no need to apply any extra function
  .void // keep the container (IO) and discards the values (puts Unit)
```

Reply:
ok, changed